packages/synapse-react-client/src/components/DataGrid/hooks/useStack.test.ts

```diff
@@ -0,0 +1,79 @@
+import { renderHook, act } from '@testing-library/react'
+import { useStack } from './useStack'
+
+describe('useStack', () => {
+  it('should initialize with an empty stack', () => {
+    const { result } = renderHook(() => useStack())
+    expect(result.current.stack).toEqual([])
+  })
+
+  it('should push items onto the stack', () => {
+    const { result } = renderHook(() => useStack())
+    act(() => {
+      result.current.push(1)
+    })
+    expect(result.current.stack).toEqual([1])
+  })
+
+  it('should return the correct size of the stack', () => {
+    const { result } = renderHook(() => useStack())
+    act(() => {
+      result.current.push(1)
+      result.current.push(2)
+    })
+    expect(result.current.size).toBe(2)
+  })
+
+  it('should handle max stack size', () => {
+    const { result } = renderHook(() => useStack<number>([], 2))
+    act(() => {
+      result.current.push(1)
+      result.current.push(2)
+      result.current.push(3)
+    })
+    expect(result.current.stack).toEqual([2, 3])
+  })
+
+  it('should pop items off the stack', () => {
+    const { result } = renderHook(() => useStack())
+    act(() => {
+      result.current.push(1)
+      result.current.push(2)
+      result.current.pop()
+    })
+    expect(result.current.stack).toEqual([1])
```

Comment:
Also verify the popped item is returned.
```suggestion
    const { result } = renderHook(() => useStack())
    
    let popped
    act(() => {
      result.current.push(1)
      result.current.push(2)
      popped = result.current.pop()
    })
    expect(popped).toEqual(2)
    expect(result.current.stack).toEqual([1])
```